core/dbt/parser/partial.py

```diff
@@ -0,0 +1,577 @@
+    # delete the patches and tests from the patch
+    def delete_schema_mssa_links(self, schema_file, dict_key, elem):
+        # find elem node unique_id in node_patches
+        prefix = key_to_prefix[dict_key]
+        elem_unique_id = ''
+        for unique_id in schema_file.node_patches:
+            if not unique_id.startswith(prefix):
+                continue
+            parts = unique_id.split('.')
+            elem_name = parts[-1]
+            if elem_name == elem['name']:
+                elem_unique_id = unique_id
+                break
+
+        # remove elem node and remove unique_id from node_patches
+        if elem_unique_id:
+            # might have been already removed
+            if elem_unique_id in self.saved_manifest.nodes:
+                node = self.saved_manifest.nodes.pop(elem_unique_id)
+                self.deleted_manifest.nodes[elem_unique_id] = node
+                # need to add the node source_file to pp_files
+                file_id = node.file_id()
+                # need to copy new file to saved files in order to get content
+                if file_id in self.new_files:
+                    self.saved_files[file_id] = self.new_files[file_id]
+                if self.saved_files[file_id]:
+                    source_file = self.saved_files[file_id]
+                    self.add_to_pp_files(source_file)
+            # TODO: should removing patches be here or with the 'merge_patch' code?
+            # remove from patches
+            schema_file.node_patches.remove(elem_unique_id)
+
+        # for models, seeds, snapshots (not analyses)
+        if dict_key in ['models', 'seeds', 'snapshots']:
+            # find related tests and remove them
+            self.remove_tests(schema_file, elem['name'])
+
+    def remove_tests(self, schema_file, name):
+        tests = self.get_tests_for(schema_file, name)
+        for test_unique_id in tests:
+            node = self.saved_manifest.nodes.pop(test_unique_id)
+            self.deleted_manifest.nodes[test_unique_id] = node
+            schema_file.tests.remove(test_unique_id)
+
+    # Create a pp_test_index in the schema file if it doesn't exist
+    # and look for test names related to this yaml dict element name
+    def get_tests_for(self, schema_file, name):
+        if not schema_file.pp_test_index:
+            pp_test_index = {}
+            for test_unique_id in schema_file.tests:
+                test_node = self.saved_manifest.nodes[test_unique_id]
+                if test_node.sources:
+                    for source_ref in test_node.sources:
+                        source_name = source_ref[0]
+                        if source_name in pp_test_index:
+                            pp_test_index[source_name].append(test_unique_id)
+                        else:
+                            pp_test_index[source_name] = [test_unique_id]
+                else:
+                    tested_node_id = test_node.depends_on.nodes[0]
```

Comment:
If I rename a `schema.yml` entry to be wrong:
```yml
seeds:
  - name: wrong_name_of_seed
    description: ...
    columns: ... # includes tests
```
And then correctly rename it to be right:
```yml
seeds:
  - name: correct_name_of_seed
    description: ...
    columns: ... # includes tests
```
I get:
```
  File "/Users/jerco/dev/product/dbt/core/dbt/parser/partial.py", line 462, in get_tests_for
    tested_node_id = test_node.depends_on.nodes[0]
IndexError: list index out of range
```

Reply:
This one is a little weird. We actually create those test nodes even when the seed is invalid. I was using the 'depends_on.nodes' to identify the node for which the tests were created, but that doesn't exist. I updated the code to check for the existence of test_node.depends_on.nodes, but that means those bogus test nodes will hang around. We might need a different way of determining what nodes a test node is connected to in the long run.

Reply:
>  but that means those bogus test nodes will hang around

I think this is the best option we have today. We have other code that disables tests if they depend on a node that is disabled or missing, and raises a user-facing warning.

Reply:
In the new partial parsing those nodes hang around even after the name of the seed has been fixed, because without something to connect them up to the badly named entry they won't be deleted. They'll go away on a non-pp run

Reply:
Ahhh ok, I follow you now. We can add that our list of partial parsing nice-to-haves. It's an annoyance, but it won't prevent people from using it in the meantime